app/routes/application.js

```diff
@@ -1,84 +1,97 @@
 const { getOwner } = Ember;
 
 export default Ember.Route.extend(preloadDataMixin, {
-
   i18n: Ember.inject.service(),
   isErrPopUpAlreadyShown: false,
   isItemUnavailable: false,
   isLoginPopUpAlreadyShown: false,
   logger: Ember.inject.service(),
   messageBox: Ember.inject.service(),
   cordova: Ember.inject.service(),
-  isMustLoginAlreadyShown: false,
 
-  _loadDataStore: function(){
-    return this.preloadData().catch(error => {
-      if (error.status === 0 || (error.errors && error.errors[0].status === "0")) {
-        this.transitionTo("offline");
-      } else {
-        this.handleError(error);
-      }
-    }).finally(() => {
-      // don't know why but placing this before preloadData on iPhone 6 causes register_device request to fail with status 0
-      if (this.session.get('isLoggedIn')) {
-        this.get("cordova").appLoad();
-      }
-    });
+  _loadDataStore: function() {
+    return this.preloadData()
+      .catch(error => {
+        if (
+          error.status === 0 ||
+          (error.errors && error.errors[0].status === "0")
+        ) {
+          this.transitionTo("offline");
+        } else {
+          this.handleError(error);
+        }
+      })
+      .finally(() => {
+        // don't know why but placing this before preloadData on iPhone 6 causes register_device request to fail with status 0
+        if (this.session.get("isLoggedIn")) {
+          this.get("cordova").appLoad();
+        }
+      });
   },
 
   init() {
     var _this = this;
-    var storageHandler = function (object) {
+    var storageHandler = function(object) {
       var currentPath = window.location.href;
-      var authToken = window.localStorage.getItem('authToken');
-      if(!authToken && !object.get('isMustLoginAlreadyShown') && !(currentPath.indexOf("login") >= 0 || currentPath.indexOf("authenticate") >= 0)) {
-        object.set('isMustLoginAlreadyShown', true);
-        object.get('messageBox').alert(object.get("i18n").t('must_login'), () => {
-          object.session.clear();
-          object.store.unloadAll();
-          object.transitionTo("login");
-        });
-      } else if(authToken && (currentPath.indexOf("login") >= 0 || currentPath.indexOf("authenticate") >= 0)) {
+      var authToken = window.localStorage.getItem("authToken");
+      if (
+        !authToken &&
+        !(
+          currentPath.indexOf("login") >= 0 ||
+          currentPath.indexOf("authenticate") >= 0
+        )
+      ) {
+        object.session.clear();
+        object.store.unloadAll();
+        object.transitionTo("login");
+      } else if (
+        authToken &&
+        (currentPath.indexOf("login") >= 0 ||
+          currentPath.indexOf("authenticate") >= 0)
+      ) {
         object.transitionTo("/");
       }
     };
-    window.addEventListener("storage", function() {
-      storageHandler(_this);
-    }, false);
+    window.addEventListener(
+      "storage",
+      function() {
```

Comment:
if you replace `storageHandler` with a lambda (`var storageHandler =(object) => { ... }`) there won't be a need for `_this` anymore

And you can do `window.addEventListener('storage', storageHandler, false)`

Not a big deal, just thought it's worth mentionning